andela/neon-ah-backend
# [#163582230] Fix bug on signup verification controller

#### What does this PR do?
Adds conditions to catch edge cases in sign up.

#### Description of Task to be completed?
- Sends a verification email to a signed up but unverified user.
- Responds appropriately to a user who is signed up and verified who wants to sign up again.
- Sends a verification email to a user who needs a new verification link

#### How should this be manually tested?
- Use git fetch to clone this branch.
- Switch to this branch in your machine.
- Run npm run dev to start the server.
- Scenario 1:
- Sign up a user at `http://localhost:3000/api/v1/auth/signup`
- Without a verifying the user, try to sign up the user again. You will get a response in this format.
`status: 'success',
              data: {
                statusCode: 200,
                message:
                  'You had started your registration process earlier. '
                  + 'Kindly check your email to complete your registration'
}`
- Scenario 2: 
-- Sign up the user
-- Verify the user by clicking the link in the email
-- Try to sign up again.
-- You will get a response in this format:
`"status": "success",
    "data": {
        "message": "You have already been registered on Author's Haven. Kindly proceed to login"
    }`

- Scenario 3:
-- Sign up a User.
-- For some reason, maybe due to expired token, the user needs a verification link.
-- Enter your email in the request body
-- Send a post request to `http://localhost:3000/api/v1/auth/resend-verification-link`
-- You will get a response in this format:
`"status": "success",
    "data": {
        "statusCode": 200,
        "message": "Kindly your check your email to verify your account"
    }`
- You can test this feature by running npm t

#### What are the relevant pivotal tracker stories?
[Link to Pivotal Tracker Story](https://www.pivotaltracker.com/n/projects/2229851/stories/163582230)
